tests/test_models_types.py

```diff
@@ -0,0 +1,195 @@
+        with self.assertRaises(AssertionError) as context:
+            GraphQLStreamfield(
+                self.field_name,
+                required=True,
+                deprecation_reason=self.deprecation_reason,
+            )()
+        assert (
+            str(context.exception)
+            == f"Argument {self.field_name} is required, cannot deprecate it."
+        )
+
+    def test_streamfield_required(self):
+        """
+        Test the GraphQLStreamfield class with required=True.
+        """
+        MyType = object()
+        field = GraphQLField(MyType, required=True)
+
+        # Assert field type is NonNull
+        assert isinstance(field.field_type, graphene.NonNull)
+
+    def test_collection_field(self):
+        """
+        Test the GraphQLCollection class.
+        """
+        MyType = GraphQLString
+        field = GraphQLCollection(
+            MyType,
+            self.field_name,
+            description=self.description,
+            deprecation_reason=self.deprecation_reason,
+        )()
+        # Use get_field_type to replicate behavior of grapple.actions.load_type_fields
+        field, field_wrapper = get_field_type(field)
+        # Assert field_wrapper type is List
+        assert isinstance(field_wrapper, graphene.List)
+
+        MyType = GraphQLSnippet
+        field = GraphQLCollection(
+            MyType,
+            self.field_name,
+            "testapp.Advert",
+            description=self.description,
+            deprecation_reason=self.deprecation_reason,
+        )()
+        # Use get_field_type to replicate behavior of grapple.actions.load_type_fields
+        field, field_wrapper = get_field_type(field)
+        # Assert field_wrapper type is QuerySetList
+        assert isinstance(field_wrapper, QuerySetList)
+
+        MyType = GraphQLForeignKey
+        field = GraphQLCollection(
+            MyType,
+            self.field_name,
+            "testapp.CustomImage",
+            description=self.description,
+            deprecation_reason=self.deprecation_reason,
+        )()
+        # Use get_field_type to replicate behavior of grapple.actions.load_type_fields
+        field, field_wrapper = get_field_type(field)
+        # Assert field_wrapper type is QuerySetList
+        assert isinstance(field_wrapper, QuerySetList)
+
+        # is_paginated_queryset should return PaginatedQuerySet
+        MyType = GraphQLForeignKey
+        field = GraphQLCollection(
+            MyType,
+            self.field_name,
+            "testapp.CustomImage",
+            is_paginated_queryset=True,
+            description=self.description,
+            deprecation_reason=self.deprecation_reason,
+        )()
+        # Use get_field_type to replicate behavior of grapple.actions.load_type_fields
+        field, field_wrapper = get_field_type(field)
+        # Assert field_wrapper type is Field and issubclass of BasePaginatedType
+        assert isinstance(field_wrapper, graphene.Field)
```

Comment:
we're using unnittest, so `self.AssertIsInstance` etc

Reply:
Fixed in https://github.com/torchbox/wagtail-grapple/pull/333/commits/c6d8a8bed8ccbb79b0e673713e67d80ec1a98712